### PR TITLE
New Telemetry Support for Luis Recognizer

### PR DESCRIPTION
**Pull request by daveta**
Adds support for IBotTelemetryClient (no new dependencies) within the Luis Recognizer.

https://github.com/daveta/analytics/blob/master/telemetry_enhancements/TelemetryEnhancements.md#telemetry-support-luis
- Properties can be added to the telemetry event per invocation 
- Class can be derived to override default behavior
- Exposed methods to help populate events when deriving

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisTelemetryConstants.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    /// <summary>
+    /// The IBotTelemetryClient event and property names that logged by default.
+    /// </summary>
+    public static class LuisTelemetryConstants
+    {
+        public const string LuisResult = "LuisResult";  // Event name
```
Comment (johnataylor) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisTelemetryConstants.cs]: could these be static readonly? (i.e. do you ever use then in a switch-case statement?)

**Comment by daveta** [comment]
Thanks @johnataylor!  
@ryanlengel mentioned some impending requests from customers - waiting until he gets back today to see what those are..

**Comment by ryanlengel** [comment]
@daveat I added comments to #1430 - they are related to the LUIS recognizer but the changes necessary are in QnA

**Comment by ryanlengel** [comment]
@daveta I added comments to #1430 - they are related to the LUIS recognizer but the changes necessary are in QnA

**Pull request merged**